specs/configuration_files_specification.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Configuration Files Specification
+layout: posts
+version: 1
+---
+
+# Configuration Files Specification
+
+## Introduction
+
+Various specifications attempt to define configuration files and file formats. This
+specification establishes where these files should be looked for, in which
+order, and how precedence, masking, extensions and overrides work.
+
+The purpose of the rules defined here is to allow OS vendors to implement the
+hermetic-usr pattern, where all vendor files are shipped in the vendor tree itself
+(`/usr/`), including configuration files with system defaults, while allowing local
+or vendor overrides without modifying the original files, for easier management. This is
+especially beneficial for image-based deployments, where the vendor tree is read-only.
+
+These rules are derived from existing real-world usage from the [systemd
+project](https://github.com/systemd/systemd) and the [libeconf
+project](https://github.com/openSUSE/libeconf).
+
+This specification is agnostic toward the actual format and content of the configuration
+files, the precise path used under each top-level hierarchy, and also toward the
+filenames and extensions, with the exception of the `.d/` notation for drop-ins
+directories. It is strongly encouraged to enforce a specific suffix for the configuration
+files, in order to disambiguate (e.g.: with backup files), but the choice of which suffix
+to use is left to each implementation.
+
+## Storage Directories and Overrides
+In order to allow shipping system defaults owned by the OS vendor, while at the same
+time letting local users or admins override those defaults, `/usr/` and `/etc/` are both
+supported for storage of configuration files, with the latter having higher priority.
+Programs must work correctly if no configuration files are found in `/etc/`.
+Optionally, `/run/` is also supported for ephemeral overrides.
+
+For example, `/usr/lib/foo/bar.conf` provides the default configuration file, but a user
+can create `/etc/foo/bar.conf` which would take precedence and completely override
+the former. If `/run/foo/bar.conf` is present and supported, it would take precedence
+over `/usr/lib/foo/bar.conf`, but not `/etc/foo/bar.conf`, which would still have the
+highest priority.
```

Comment:
Perhaps follow the order of precedence?

```suggestion
For example, `/usr/lib/foo/bar.conf` provides the default configuration file. 
If `/run/foo/bar.conf` is present and supported, it would take precedence
over `/usr/lib/foo/bar.conf`.
Finally, a user can create `/etc/foo/bar.conf` which would take precedence
and completely override both.
```